### ISSUE TITLE
Use prefix to distinghuish aggregate functions?

### ISSUE THREAD

**Issue by FransKnibbe** [opengeospatial/ogc-geosparql#209]
GeoSPARQL adds aggregate functions. But their names can conflict with non-aggregate functions. That is why there now is a function `geof:union2`, which is the aggregate form of `geof:union`.
Would it be a good idea to indicate aggregrate functions by a common prefix? That could make aggregate functions more recognisable, both as individuals and as a group. It could also help resolve naming conflicts. Grouping functions with a prefix is already done in GeoSPARQL, see for example `geo:sfEquals` and `geo:ehEquals`. Similarly, `union2` could become something like `agUnion`.

**Comment by situx** [comment]
Initially, I did that, but then the opinion was that one namespace for functions would be sufficient.
What we could consider though is just renaming the union2 function?

**Comment by FransKnibbe** [comment]
> Initially, I did that, but then the opinion was that one namespace for functions would be sufficient.

Yes, but prefixes are different from namespaces, aren't they? There are already different function prefixes in use: "eh", "rcc", "sf". Adding a new prefix would be continuing the current practice.

> What we could consider though is just renaming the union2 function?

It is not unimaginable that new naming conflicts will occur when new functions are added in the future. Using prefixes for function categories solves that general problem.

**Comment by situx** [comment]
Oh, sorry, I misunderstood you there. Sure one could think about adding prefixes in function names.

**Comment by nicholascar** [comment]
Use of "prefix" here is confusing but I get the point. Indeed `geof:ag[function-name]` might work well.

**Comment by jabhay** [comment]
prefix `agg` agreed.

**Comment by FransKnibbe** [comment]
@situx: 01-Preamble.adoc also contains function names,

**Comment by situx** [comment]
@FransKnibbe Thanks for spotting this. It is fixed in the pull request of @nicholascar with the editorial changes
